### PR TITLE
Change to template - Environment Variables / Parameterized the template

### PR DESCRIPTION
**Pull request by ChillBill77**
Parameterized the template
Add some comments/additions to underlying files

added to README.md

Please check the following contents for validity as some variables need to be defined upfront.

- [ ] .env // This files hold the main parts
- [ ] ./config/serve.json // This file requires a service port of the app to be defined

### CODE REVIEW

File: templates/service-template/docker-compose.yml [hunk 1]
```diff
@@ -30,21 +31,21 @@ services:
       start_period: 10s # Time to wait before starting health checks
     restart: always
 
-  # SERVICE
-  SERVICE:
-    image: lscr.io/linuxserver/SERVICE:latest # Image to be used
-    network_mode: service:tailscale-SERVICE # Sidecar configuration to route SERVICE through Tailscale
```
Comment (crypt0rr) [inline templates/service-template/docker-compose.yml:36]: This only works when `service:` is present. Please alter the `network_mode: tailscale` to `network_mode: service:tailscale`

File: templates/service-template/docker-compose.yml [hunk 2]
```diff
@@ -1,27 +1,28 @@
...
     cap_add:
       - net_admin # Tailscale requirement
       - sys_module # Tailscale requirement
     ports:
-      - 0.0.0.0:SERVICEPORT:SERVICEPORT # Binding port SERVICEPORT to the local network - may be removed if only exposure to your Tailnet is required
+      - 0.0.0.0:${SERVICEPORT}:${SERVICEPORT} # Binding port ${SERVICE}PORT to the local network - may be removed if only exposure to your Tailnet is required
     # If any DNS issues arise, use your preferred DNS provider by uncommenting the config below
-    # dns: 
-    #   - 1.1.1.1 
+    dns:
```
Comment (crypt0rr) [inline templates/service-template/docker-compose.yml:24]: Please comment out the DNS option, this should not be default but optional.

Comment (ChillBill77) [reply templates/service-template/docker-compose.yml]: Done.

**Comment by ChillBill77** [comment]
Made minor changes to the docker-compose.

**Comment by crypt0rr** [comment]
@ChillBill77 Please check the other comments as well

File: templates/service-template/config/serve.json [hunk 3]
```diff
@@ -8,7 +8,7 @@
     "${TS_CERT_DOMAIN}:443": {
       "Handlers": {
         "/": {
-          "Proxy": "http://127.0.0.1:PORTNUMBER"
+          "Proxy": "http://127.0.0.1:${SERVICEPORT}"
```
[the remark this answers is not in the document]
Comment (crypt0rr) [reply templates/service-template/config/serve.json:11]: @ChillBill77 please fix

File: templates/service-template/docker-compose.yml [hunk 4]
```diff
@@ -1,27 +1,28 @@
...
+# Make sure you have updated/checked the .env file with the correct variables. 
+# All the ${ xx } need to be defined there.
   # Tailscale Sidecar Configuration
-  tailscale-SERVICE:
+  tailscale:
     image: tailscale/tailscale:latest # Image to be used
-    container_name: tailscale-SERVICE # Name for local container management
-    hostname: SERVICE # Name used within your Tailscale environment
+    container_name: ts-${SERVICE} # Name for local container management
+    hostname: ts-${SERVICE} # Name used within your Tailscale environment
```
[the remark this answers is not in the document]
Comment (crypt0rr) [reply templates/service-template/docker-compose.yml:8]: @ChillBill77 please fix

**Review by crypt0rr — CHANGES_REQUESTED** [review]

File: templates/service-template/docker-compose.yml (hunk 4 above)
[the remark this answers is not in the document]
Comment (ChillBill77) [reply templates/service-template/docker-compose.yml:8]: Already fixed in earlier commit

**Pull request merged**